parcels/codegenerator.py

```diff
@@ -864,42 +865,43 @@ def __init__(self, fieldset, ptype=None):
     def generate(self, funcname, field_args, const_args, kernel_ast, c_include):
         ccode = []
 
+        pname = self.ptype.name + 'p'
+
         # Add include for Parcels and math header
         ccode += [str(c.Include("parcels.h", system=False))]
         ccode += [str(c.Include("math.h", system=False))]
         ccode += [str(c.Assign('double _next_dt', '0'))]
         ccode += [str(c.Assign('size_t _next_dt_set', '0'))]
 
-        # Generate type definition for particle type
-        vdecl = []
-        for v in self.ptype.variables:
-            if v.dtype == np.uint64:
-                vdecl.append(c.Pointer(c.POD(np.void, v.name)))
-            else:
-                vdecl.append(c.POD(v.dtype, v.name))
-
+        # Generate type definition for particle struct type
+        vdeclp = [c.Pointer(c.POD(v.dtype, v.name)) for v in self.ptype.variables]
+        ccode += [str(c.Typedef(c.GenerableStruct("", vdeclp, declname=pname)))]
+        # Generate type definition for single particle type
+        vdecl = [c.POD(v.dtype, v.name) for v in self.ptype.variables if v.dtype != np.uint64]
```

Comment:
Good catch! That might actually also be related the failing CI on Windows. which is also about `uint64` (although then of course it's strange that it does work on macos and linux)

Reply:
I just recognized that the variables are now named differently, so they don't override one another. That clears up my confusion.

This, this looks a bit confusing, because now, wouldn't we have two fields (instead of one) for each variable, thus being either wasteful or ambiguous for the c-kernel ?

Reply:
Good point, something to explore

In any case, removing the `if v.dtype != np.uint64` part of the statement did not change anything in the unit tests (see commit 67bf24b)

Reply:
so, thanks to @angus-g feedback from cgen, I'm thinking again: would it inappropriate to apply this differentiation also for `uint32` ? most of the other variables are anyway still floating-point metrics, so that wouldn't really confuse it. I'd try that out.